packages/core/utils/src/common/validate-module-name.ts

```diff
@@ -0,0 +1,25 @@
+/**
+ * Validates the module name to be variable safe. Since we generate
+ * a lot of code, types under the hood using the module name we
+ * have ensure that each module name is variable safe.
+ *
+ * Ofcourse, we can transform the module name to a variable safe value,
+ * but that might result into naming conflicts. For example: There are
+ * two module named as
+ *
+ * - sanity-products
+ * - sanity_products
+ *
+ * After transforming them, they will endup with the same output. This is
+ * a very simple example, but cases like these will lead to naming
+ * conflicts, so its better to use the names as it is and restrict
+ * them to be variable safe
+ */
+const RE = /^[a-zA-Z_][0-9a-zA-Z_]*$/
+export function validateModuleName(name: string) {
+  if (!RE.test(name)) {
+    throw new Error(
+      `Invalid module name "${name}". Module names must be alpha numeric and may contain an underscore`
+    )
```

Comment:
suggestion: Should we log error and skip the registration of the module to avoid the breaking change? I

Reply:
You mean still register the module?

Wouldn't that eventually fail when you will try to run migrations or generate GQL schema?

Reply:
Can we say with certainty that module names that don't follow these rules would have led to errors at some point, e.g. for running migrations? 

If that's the case, I'd be fine to proceed with this. But if not, I think we should log + skip the registration.

Reply:
I can run some tests locally

Reply:
Yes, I can confirm that using any of the unallowed values doesn't work. I cannot start the server

Reply:
Great, thanks for testing. Let's merge as is